### DISCUSSION TITLE
How to find out which version of SQLPage is running ?

### DISCUSSION THREAD

**Discussion by Zorg77127** [sqlpage/SQLPage#280] [General]
Hi Community,

All is in the title...
Is it possible to know sqlpage.exe release ? for example sqlpage -v or better a function like sqlpage_release() ? Maybe it already exists, and I missed something...
Thanks in advance !

**Comment by lovasoa** [comment]
Hello !

  SQLPage currently does not accept any command-line flag, such as `-v`. However :

 - The default *Built with SQLPage* [shell footer]() shows the SQLPage version on hover:
   - ![image](https://github.com/lovasoa/SQLpage/assets/552629/16b8abaa-ce86-48a7-8939-6e134ffa1f45)
 - The [`sqlpage.version()`](https://sql.ophir.dev/functions.sql?function=version#function) function returns the current SQLPage version.
 - The HTTP response headers contain the SQLPage version string
   - ```
     ❯ curl -I localhost:8080
      HTTP/1.1 200 OK
      transfer-encoding: chunked
      server: sqlpage v0.20.2
      content-type: text/html; charset=utf-8
      content-security-policy: script-src 'self' https://cdn.jsdelivr.net
      date: Tue, 02 Apr 2024 10:44:30 GMT
      ```
 - The default home page, displayed when no `index.sql` is present, includes the running SQLPage version
   - ![image](https://github.com/lovasoa/SQLpage/assets/552629/c3a0c69e-f16f-40ae-844e-9ff2bed04880)
 - In [custom components](https://sql.ophir.dev/custom_components.sql) written in handlebars, `{{buildinfo 'CARGO_PKG_VERSION'}}` can be used to include the running SQLPage version in a page.

**Reply by Zorg77127** [reply]
that's perfect, thank you very much !
